lib/presentation/shared_widgets/custom_app_bars/scrollable_app_bar.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:collaction_app/presentation/themes/constants.dart';
+import 'package:flutter/material.dart';
+
+class ScrollableAppBar extends StatefulWidget implements PreferredSizeWidget {
+  final String title;
+  final bool centerTitle;
+  final Color backgroundColor;
+  final ElevatedButton? leading;
+  final List<ElevatedButton>? actions;
+  final TextStyle? titleTextStyle;
+  final ScrollController pageScrollController;
+  final double? elevation;
+  final double minElevation;
+  final double maxElevation;
+  const ScrollableAppBar(
+      {Key? key,
+      this.title = '',
+      this.centerTitle = true,
+      this.backgroundColor = kSecondaryColor,
+      this.titleTextStyle,
+      this.leading,
+      this.actions,
+      this.elevation,
+      this.minElevation = 0.0,
+      this.maxElevation = 2.0,
+      required this.pageScrollController})
+      : super(key: key);
+
+  @override
+  ScrollableAppBarState createState() => ScrollableAppBarState();
+
+  @override
+  Size get preferredSize => const Size(double.infinity, 56.0);
+}
+
+@visibleForTesting
+class ScrollableAppBarState extends State<ScrollableAppBar> {
+  late final ScrollController scrollController;
+
+  late double currentElevation;
+
+  @override
+  void initState() {
+    super.initState();
+    scrollController = ScrollController();
+
+    currentElevation = widget.elevation ?? widget.minElevation;
+
+    widget.pageScrollController.addListener(() {
+      if (widget.pageScrollController.position.pixels <=
+          scrollController.position.maxScrollExtent) {
+        scrollController.jumpTo(widget.pageScrollController.position.pixels);
+      }
+
+      if (widget.pageScrollController.position.pixels >
+          widget.pageScrollController.position.minScrollExtent) {
+        if (widget.elevation == null) {
+          if (currentElevation == widget.minElevation) {
+            setState(() {
+              currentElevation = widget.maxElevation;
+            });
+          }
+        }
+      } else if (widget.pageScrollController.position.pixels ==
```

Comment:
Same here, remove the innermost if statement, change this line to be something like:
`} else if (widget.elevation == null && widget.pageScrollController.position.pixels == widget.pageScrollController.position.minScrollExtent) {`
and then just have the `setState(...)` call inside it.